DB 연결 및 User model, schema 생성

allmie opened the pull request:
## 개요

- DB 연결 및 User model, schema 생성
 
## 작업 내용

- 'study-with' DB 생성
- User schema, model 생성
   - nickname, email 항목만 필수 값, 나머지 값은 옵션 값으로 설정
   - avartarImg
      - avatar -> avatarImg 이름 변경
   - tel
      - 추가
   - job
   - jobLevel
     - status -> jobLevel 이름 변경
   - availableLocation, availableWeek, availableTime
     - 회의 관련 값 추가
   - learningPeriod
     - learning -> learningPeriod 이름 변경
   - interestSkills
     - skills -> interestSkills 이름 변경
   - joinDate
   - intro
   - portfolio
   - role
     - objectID type 사용
   - receivedLike
     - objectID type 사용
   - token

* ##### 차주에 myPage, people 등 user 정보가 들어가는 페이지 markup 수정이 필요할 것으로 보임

## 스크린샷

#### DB 연결 확인
![db 연결 성공 확인1](https://user-images.githubusercontent.com/31345532/116504830-38997600-a8f4-11eb-8c0c-7beeb2db18a2.png)

#### 저장한 user 값 확인
![db 데이터 저장](https://user-images.githubusercontent.com/31345532/116504825-36371c00-a8f4-11eb-8e39-17e358993bd0.PNG)

#### Atals DB
![image](https://user-images.githubusercontent.com/31345532/116785853-b026fb00-aad6-11eb-9905-778bd4e0e09f.png)

allmie committed:
feat: study-with DB 연결

allmie committed:
feat: User model, schema 생성

- nickname, email만 필수 item으로 설정

allmie committed:
feat: User schema 항목 추가

- received like(받은 좋아요 수) 추가

allmie committed:
fix: User schema 값 이름 수정

- availableWeeik -> availableWeek

hi-jade88 commented on server/db.ts:
```diff
@@ -0,0 +1,29 @@
+import mongoose from 'mongoose';
+
+mongoose.connect('mongodb://127.0.0.1:27017/study-with', {
```
mongoURI는 env파일에 넣는게 보안 상 좋을꺼같습니다.

allmie replied:
보안상 맞긴 한데, 
코드 공유를 위해 env 파일 없이 작성하는 거 아니었나요??

allmie replied:
수정했습니다

hi-jade88 commented on server/app.ts:
```diff
@@ -1,31 +1,32 @@
 import express from 'express';
+import './db';
+import User from './model/User';
```
require('dotenv').config();
const { PORT, MONGO_URI } = process.env;
이렇게 추가하고 .env 파일에 PORT. MONGO_URI 적어두고 gitignore하면 좋을꺼같아요

hi-jade88 commented on server/app.ts:
```diff
@@ -1,31 +1,32 @@
+app.use('/api/users', UserRouter);
+app.use('/api/project', ProjectRouter);
 
 const port = process.env.PORT || 5000;
```
여기 줄에 포트 번호가 코드에 공개되는 것이 보안 상 안 좋을꺼같으니 지우고 위env파일에서 받아서 쓰는게 어떨까요

hi-jade88 commented on server/model/User.ts:
```diff
@@ -0,0 +1,45 @@
+
+const userSchema = new mongoose.Schema({
+  avartarImg: String,
+  nickname: {
```
unique값 추가해주세오

hi-jade88 replied:
닉네임이랑 이메일 전화번호는 unique: true trim: true넣는게 좋을꺼같아요

allmie replied:
수정했습니다

hi-jade88 commented on server/model/User.ts:
```diff
@@ -0,0 +1,45 @@
+  learningPeriod: String,
+
+  interestSkills: [String],
+  joinDate: {
```
이걸 따로 두지말고 timestamps:true값을 넣는게 어떨까요

allmie replied:
수정했습니다

hi-jade88 commented on server/model/User.ts:
```diff
@@ -0,0 +1,45 @@
+  //   ref: 'Like',
+  // },
+  token: String,
+});
```
여기에 timestamps 넣으면 자동으로 생성날짜랑 업데이트 날짜가 찍혀서 저장됩니닿

hi-jade88 commented on server/db.ts:
```diff
@@ -0,0 +1,29 @@
+const handleOpen = () => {
+  console.log('Connected to Study-with DB!!');
+};
+const handleError = (error: any) =>
```
any type 보다는 명확한 타입을 명시하는 것이 좋고 아마 @types/mongoose를 설치해야할꺼에요. 아마 설치하고 Error 쓰면 될꺼같아요. 그리고 handleError함수로 따로 부르는 경우에 초반에 처음 연결 때 에러를 잡지 못해서 위에 connet하고 catch해서 잡아야한다고 들었는데 한번 고려해주세요.

allmie replied:
error를 특정 type으로 명시 할 수 있나요?? 

error 타입은 찾아보고 해결하면 수정 하겠습니다.

allmie replied:
@types/mongoose 문서에서도 (err: any)로 사용하고 있어서 일단 그대로 두었습니다.

추후에 확인 되면 수정하겠습니다

hi-jade88 reviewed (commented):
schema interface도 따로 파일 만들어서 관리해야 할거같습니닿

allmie committed:
feat: User schema 속성 추가

- nickname, email, tel: trim, unique 속성 추가
- timestamp 추가, joinDate 제거

allmie committed:
feat: User schema, model interface 생성

- User schema, model interface 생성
- db 연결 then-catch 방식으로 변경

allmie committed:
feat: User schema, model interface 수정

User schema
- learningPeriod 삭제
- job, jobLevel -> position, positionLevel 변경
- receivedLike default 값 추가
- join 생성(참여하고 있는 Project id)

User interface
- join, portpolio option 값으로 변경

allmie committed:
fix: User schema의 tel 속성 변경

- "unique: true"일 때 required와 상관 없이 해당 값을 필요로 함
- tel 필수 입력 옵션이 아니기 때문에 unique 속성 제거, interface에서 선택 옵션으로 변경

allmie committed:
feat: 클라우드 database 사용

- 로컬 mongoDB -> MongoDB Atlas 변경

allmie committed:
feat: DB서버 포트, URL 보안

- dotenv를 사용해 DB 서버와 URL 보호
- 코드 공유를 위해 .env파일 .gitignore에서 제외

dhgu-dev commented on server/model/User/User.ts:
```diff
@@ -0,0 +1,56 @@
+        ref: 'Project',
+      },
+    ],
+    portfolio: [String],
```
portfolio에는 url 배열이 들어갈테니 타입을 String[]로 지정해야 할 것 같은데 [String] 표기가 맞는 건가요?
(상단의 interestSkills도 같은 방법으로 표기되어 있습니다)

hi-jade88 commented:
이미 main에 머지된듯 합니다.
